py/farm_ng/track/utils.py

```diff
@@ -45,3 +54,314 @@ def update_filter_track(track_path: Path) -> None:
     filter_track: FilterTrack = proto_from_json_file(track_path, FilterTrack())
     track: Track = filter_track_to_track(filter_track)
     proto_to_json_file(track_path, track)
+
+
+class TrackBuilder:
+    """A class for building tracks."""
+
+    def __init__(self, clients: dict[str, EventClient] | None = None, start: Pose3F64 | None = None) -> None:
+        """Initialize the TrackBuilder."""
+        self.clients = clients if clients is not None else {}
+        self._start: Pose3F64()
+        self.track_waypoints: list[Pose3F64] = []
+        self._segment_indices: list[int] = [0]
+        self._loaded: bool = False
+
+        if start is not None:
+            self._start = start
+            self.track_waypoints = [start]
```

Comment:
Can we just require a start on creation?

Other methods assume there are already waypoints (e.g., `segment_poses: list[Pose3F64] = [self.track_waypoints[-1]]`, and being able to cause a crash by calling these methods after the default constructor seems unsafe.

```suggestion
    def __init__(self, start: Pose3F64, clients: dict[str, EventClient] | None = None) -> None:
        """Initialize the TrackBuilder."""
        self.clients = clients if clients is not None else {}
        self._start: Pose3F64() = start
        self.track_waypoints: list[Pose3F64] = [start]
        self._segment_indices: list[int] = [0]
        self._loaded: bool = False
```

Reply:
Sure, we can require a start on creation. I created a function in the `main.py` example to query the current robot pose, and modified the class in this commit: 3ca6744c813c1cb28c62234e2c99f0ca8846c445